### PR TITLE
[16.0][MIG] account_statement_import: Migrate to version 16.0

### PR DESCRIPTION
**Pull request by sonhd91**
- `view_bank_statement_line_form` was [removed](https://github.com/odoo/odoo/commit/4fb0933986534e31337aa1dce3478a67deea72de#diff-f3049e00919d62bbf6b11a37517e834396157646af188640bf7763122fe92957) from odoo 16
- Change module name and it's content from `account_statement_import` to `account_statement_import_file`
- Port-forward
* https://github.com/OCA/bank-statement-import/pull/482
* https://github.com/OCA/bank-statement-import/pull/522
* https://github.com/OCA/bank-statement-import/pull/527

### CODE REVIEW

**Comment by florian-dacosta** [comment]
Thanks for the work.
One issue though, a refactore was done in version 14, with the extraction of a base module in order to share code between account_statement_import and account_statement_import_online.
I guess it was not ported to v14 and is quite recent, but I believe it is still relevant for v16 and later.
Can you have a look and take it into account ? here is the PR on v14 : https://github.com/OCA/bank-statement-import/pull/482

@alexis-via You confirm it is still needed, right ?

**Comment by alexis-via** [comment]
Yes, I confirm that !

**Comment by sonhd91** [comment]
> Thanks for the work. One issue though, a refactore was done in version 14, with the extraction of a base module in order to share code between account_statement_import and account_statement_import_online. I guess it was not ported to v14 and is quite recent, but I believe it is still relevant for v16 and later. Can you have a look and take it into account ? here is the PR on v14 : #482
> 
> @alexis-via You confirm it is still needed, right ?

I just updated the branch with pr#482 in it, please check again

File: account_statement_import_file_reconciliation_widget/wizards/account_statement_import.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class AccountStatementImport(models.TransientModel):
+    _inherit = "account.statement.import"
+
+    def import_file_and_reconcile_button(self):
+        """Process the file chosen in the wizard, create bank statement(s)
+        and jump directly to the reconcilition widget"""
+        result = self._import_file()
+        statements = self.env["account.bank.statement"].browse(result["statement_ids"])
+        statements.button_post()
```
Comment (RodrigoBM) [inline account_statement_import_file_reconciliation_widget/wizards/account_statement_import.py]: button_post deleted in this commit https://github.com/OCA/OCB/commit/4fb0933986534e31337aa1dce3478a67deea72de#diff-96c5c4fc954f9deb57474d052d9c59b68cc41b610205d342949727ece08ac404L414 in v16

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

File: account_statement_import_file_reconciliation_widget/wizards/account_statement_import.py (hunk 1 above)
Comment (sonhd91) [reply account_statement_import_file_reconciliation_widget/wizards/account_statement_import.py]: Hello, I just the branch, please check again, thank you

**Comment by MiquelRForgeFlow** [comment]
account_statement_import_base is already merged. Please rebase :S

**Comment by alexis-via** [comment]
I propose to switch to my new PR https://github.com/OCA/bank-statement-import/pull/575 that migrate account_statement_import_file

**Comment by sonhd91** [comment]
> I propose to switch to my new PR #575 that migrate account_statement_import_file

Can you please help me note some different between these 2 PR ? This one migrate from version 15.0

**Pull request closed without merging**